swap/swap.go

```diff
@@ -142,50 +143,57 @@ func New(dbPath string, prvkey *ecdsa.PrivateKey, backendURL string, params *Par
 	// initialize the balances store
 	stateStore, err := state.NewDBStore(filepath.Join(dbPath, "swap.db"))
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: %s", err)
+		return nil, fmt.Errorf("swap init error: error initializing statestore: %v", err)
 	}
 	if params.DisconnectThreshold <= params.PaymentThreshold {
 		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", params.DisconnectThreshold, params.PaymentThreshold)
 	}
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
+		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %v", backendURL, err)
 	}
-
-	// we may not need this check, and we could maybe even get rid of this constant completely
-	if params != nil && params.InitialDepositAmount == 0 {
-		// need to prompt user for initial deposit amount
-		// if 0, can not cash in cheques
-		prompter := console.Stdin
-
-		// ask user for input
-		input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
-		if err != nil {
-			return nil, err
-		}
-		// check input
-		val, err := strconv.ParseInt(input, 10, 64)
+	networkID, err := backend.NetworkID(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("swap init error: error retrieving networkID from backendURL: %v", err)
+	}
+	var usedBeforeAtNetwork uint64
+	var usedBefore bool
+	err = stateStore.Get(usedBeforeAtNetworkKey, &usedBeforeAtNetwork)
+	// error reading from database
+	if err != nil && err != state.ErrNotFound {
+		return nil, fmt.Errorf("swap init error: error querying usedBeforeAtNetwork from statestore: %v", err)
+	}
+	// read from database
+	if err == nil {
+		usedBefore = true
```

Comment:
Is this safe enough? What if the state store went away?

That would be of course a user error. And in that case all balances and cheques would be gone anyway, which means that the node starts afresh.

I am just asking you to think of what this could mean for the complete initialization process here. I guess there is no problem whatsoever, please confirm.

Reply:
If the user corrupts it's state store many things can go wrong. Let's say it would only delete the entry about the networkID. In this case, the user might connect on a different network, where he will (among other potential errors) not be able to cash the cheques which should be cashable according to the statestore.